### PR TITLE
api: support unicode filenames

### PR DESCRIPTION
**Pull request by jacquerie**
Minimal reproduction for a bug that popped out from our Sentry: https://travis-ci.org/inveniosoftware/invenio-records-files/jobs/306347184#L956.

### CODE REVIEW

**Comment by jacquerie** [comment]
Now this PR also implements a workaround for the issue we saw in production, but IMO this hints that there could be more places were unicode is not handled properly.

File: invenio_records_files/api.py [hunk 1]
```diff
@@ -153,8 +154,11 @@ def __next__(self):
 
     def __contains__(self, key):
         """Test if file exists."""
+        if not isinstance(key, text_type):
```
Comment (lnielsen) [inline invenio_records_files/api.py]: Is this decoding needed? I.e. it's not enough to just remove the ``str()`` call?

Comment (jacquerie) [reply invenio_records_files/api.py]: I was utterly convinced that one needed to have `unicode` objects to use SQLalchemy's `filter`... I was wrong!

**Comment by lnielsen** [comment]
Thanks for the PR

**Pull request merged**